feat-channels/src/main/java/dev/baseio/slackclone/uichannels/views/SlackAllChannels.kt

```diff
@@ -6,17 +6,20 @@ import dev.baseio.slackclone.chatcore.data.ExpandCollapseModel
 import dev.baseio.slackclone.uichannels.R
 import dev.baseio.slackclone.uichannels.SlackChannelVM
 import androidx.compose.runtime.*
+import androidx.lifecycle.compose.ExperimentalLifecycleComposeApi
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import dev.baseio.slackclone.chatcore.data.UiLayerChannels
 
+@OptIn(ExperimentalLifecycleComposeApi::class)
 @Composable
 fun SlackAllChannels(
   onItemClick: (UiLayerChannels.SlackChannel) -> Unit = {},
   channelVM: SlackChannelVM = hiltViewModel(),
   onClickAdd: () -> Unit
 ) {
   val recent = stringResource(R.string.channels)
-  val channelsFlow = channelVM.channels.collectAsState()
-  val channels by channelsFlow.value.collectAsState(initial = listOf())
+  val channelsFlow = channelVM.channels.collectAsStateWithLifecycle()
+  val channels by channelsFlow.value.collectAsStateWithLifecycle(initialValue = listOf())
```

Comment:
better to use emptyList() instead of listOf() at all places

Reply:
@AndroidPoet

Reply:
@Anmol92verma listof() replaced with emptyList().

Reply:
Thanks, @Anmol92verma, for accepting this pull request.